### PR TITLE
Lua API Documentation Cleanups

### PR DESCRIPTION
**Pull request by rubenwardy**
The aim of the PR is to increase the readability of lua_api.txt, and to place everything into nicer sections

- [x] Add game dir structure docs
- [x] Put all mod and game info into their relevant sections
- [x] Remove incorrect mod location info
- [x] For registered id section, make main heading match contents (ie: aliases makes sense in a reg.id. section, but not in a naming conventions for reg.id. section)
- [x] Use list rather than headings for texture modifiers - increases readability imo, and is also more consistent
- [ ] Put simple sound spec into definitions of simple things
- [ ] Clean up nodes, paramtype, and draw type sections

### CODE REVIEW

File: doc/lua_api.txt [hunk 1]
```diff
@@ -140,14 +152,13 @@ Mod directory structure
     |   |-- sounds
     |   |-- media
     |   `-- <custom data>
-    `-- another
-
+    `-- another`
```
Comment (SmallJoker) [inline doc/lua_api.txt]: This list style should be changed a bit. I think this looks a bit strange in a graphical markdown view.

Comment (rubenwardy) [inline doc/lua_api.txt]: I agree. It also breaks my highlighter. Maybe something like:

```
C:.
├───AdminScripts
├───iissamples
├───mailroot
│   ├───Badmail
│   ├───Drop
│   ├───Mailbox
│   ├───Pickup
│   ├───Queue
│   ├───Route
│   └───SortTemp
├───Scripts
└───wwwroot
    ├───aspnet_client
    │   └───system_web
    │       └───1_1_4322
    │           └───_vti_cnf
    ├───images
    ├───_private
    └───_vti_log
```

Comment (SmallJoker) [inline doc/lua_api.txt]: Perhaps add a space between the last dash and name but otherwise LGTM.

**Comment by paramat** [comment]
Please focus on readability in .txt form, not MD graphical form. The file is viewed as .txt far more often than MD graphical.
Why is this document formatted for MD at all? It is viewed as normal text 99% of the time and the dded MD symbols make it messy.
Please can we remove all MD format changes? that was a mistake.

File: doc/lua_api.txt (hunk 1 above)
Comment (paramat) [inline doc/lua_api.txt]: Yes please format in plain text as in rubenwardy'ds example.

**Comment by rubenwardy** [comment]
Part of my changes is to try and make it easier to read - by removing pointless \` ticks, and using a list rather than annoying headings

**Comment by paramat** [comment]
Good, might as well remove all MD formatting then.

**Comment by rubenwardy** [comment]
Nope, it's still useful to make it consistent and [also allows HTML conversion](http://rubenwardy.com/minetest_modding_book/lua_api.html)

**Comment by paramat** [comment]
Ok allowing to link to a section is useful, that should be kept of course.

**Comment by paramat** [comment]
Also, if we were not using `pos` we would be using 'pos' which is not much different, so that is ok too.
Looking at your modding book i think you're the ideal person to work on this.

**Comment by numberZero** [comment]
looks nice. just don’t forget there is `vim`, it uses the markup too ;) (although it can’t recognize the file type automatically)

**Comment by rubenwardy** [comment]
closing as heavily WIP, and probably needs restarting

**Pull request closed without merging**